fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -2924,4 +2922,30 @@ private boolean allSearchParmsAreGlobal(List<QueryParameter> queryParms) {
             log.exiting(CLASSNAME, METHODNAME);
         }
     }
-}
+    
+    @Override
+    public boolean isUpdateCreateEnabled() {
+        return this.updateCreateEnabled;
+    }
+
+    @Override
+    public Future<PayloadKey> storePayload(Resource resource, String logicalId, int newVersionNumber, Instant lastUpdated) throws FHIRPersistenceException {
+        if (payloadPersistence != null) {
+            InputOutputByteStream ioStream = new InputOutputByteStream(DATA_BUFFER_INITIAL_SIZE);
+            try (GZIPOutputStream zipStream = new GZIPOutputStream(ioStream.outputStream())) {
```

Comment:
I've seen guidance for COS to avoid gzipping the content, but I think its more about being able to query over that data using Spark SQL and the like.  Still, I'm wondering if we should let the payloadPersistence object decide whether to gzip or not...

Reply:
I've seen the same guidance as Lee. 

1 - I thought this was being zipped at a lower level in persistence-cos / persistence-cassandra already? There is definitely a duplicate pattern used (are these being double zipped and unzipped? 

2 - it's a common pattern again, perhaps push down to fhir-model